datadreamer/prompt_generation/profanity_filter.py

```diff
@@ -0,0 +1,180 @@
+
+    Attributes:
+        seed (Optional[float]): Seed for randomization.
+        device (str): Device to run the language model on ('cuda' for GPU, 'cpu' for CPU).
+        use_lm (bool): Whether to use a language model for checking text safety.
+        bad_words (List[str]): List of bad words to filter from texts.
+        model (AutoModelForCausalLM): The pre-trained causal language model for checking text safety.
+        tokenizer (AutoTokenizer): The tokenizer for the pre-trained language model.
+
+    Methods:
+        set_seed(seed): Sets the random seed for consistent prompt generation.
+        _init_lang_model(): Initializes the language model and tokenizer.
+        _contains_bad_words(texts): Checks if a list of texts contain bad words.
+        _check_lm_safety(text): Checks if a text is safe using a language model.
+        is_safe(classes): Checks if a list of classes is safe.
+        release(empty_cuda_cache): Releases the model and optionally empties the CUDA cache.
+    """
+
+    LLM_PROMPT = """You are Qwen, created by Alibaba Cloud. You are a helpful assistant who classifies the classes as appropriate or inappropriate. Inappropriate classes are those that directly relate to drugs, hate, racism, harassment, nudity, sexual or offensive words. Here are inappropriate examples:
+- 'ass',
+- 'a**',
+- 'bitch',
+- 'pussy',
+- and 'f**k'.
+
+Otherwise, the classes are considered appropriate. They can talk about people, characters, animals, nature, history, human conflicts, and so on. Some acceptable examples are:
+- 'cat',
+- 'angry barking dog',
+- 'alien',
+- 'dracula',
+- 'war',
+- 'soldier',
+- 'pluto',
+- 'sun',
+- and 'mercury.'
+
+Respond 'inappropriate' if the classes are unacceptable, otherwise respond with 'appropriate'."""
+
+    def __init__(
+        self,
+        device: str = "cuda",
+        use_lm: bool = False,
+        seed: Optional[float] = 42,
+    ) -> None:
+        """Initializes the ProfanityFilter with parameters."""
+        self.seed = seed
+        if seed is not None:
+            self.set_seed(seed)
+        self.device = device
+        self.use_lm = use_lm
+        if self.use_lm:
+            self.model, self.tokenizer = self._init_lang_model()
+
+    @staticmethod
+    def set_seed(seed: int) -> None:
+        """Sets the random seed for consistent prompt generation.
+
+        Args:
+            seed (int): The random seed.
+        """
+        random.seed(seed)
+        torch.manual_seed(seed)
+        torch.cuda.manual_seed_all(seed)
+
+    def _init_lang_model(self) -> Tuple[AutoModelForCausalLM, AutoTokenizer]:
+        """Initializes the language model and tokenizer for prompt generation.
+
+        Returns:
+            tuple: The initialized language model and tokenizer.
+        """
+        model_name = "Qwen/Qwen2.5-1.5B-Instruct"
+        model = AutoModelForCausalLM.from_pretrained(
+            model_name, torch_dtype="auto", device_map=self.device
```

Comment:
CPU and CUDA initialization isn't different here, right? (like in prompt generators)